File: library/src/main/java/com/mikepenz/materialdrawer/model/interfaces/Nameable.java
package com.mikepenz.materialdrawer.model.interfaces;

import com.mikepenz.materialdrawer.holder.StringHolder;

/**
 * Created by mikepenz on 03.02.15.
 */
public interface Nameable<T> {
    T withName(String name);

    T withName(int nameRes);

    T withName(StringHolder name);

    StringHolder getName();
}

File: library/src/main/java/com/mikepenz/materialdrawer/model/interfaces/Nameable.kt
package com.mikepenz.materialdrawer.model.interfaces

import androidx.annotation.StringRes
import com.mikepenz.materialdrawer.holder.StringHolder

/**
 * Defines a [IDrawerItem] with support for defining a name
 */
interface Nameable {
    var name: StringHolder?
}

@Deprecated("Please consider to replace with the actual property setter")
fun <T : Nameable> T.withName(name: String?): T {
    this.name = StringHolder(name)
    return this
}

@Deprecated("Please consider to replace with the actual property setter")
fun <T : Nameable> T.withName(@StringRes name: Int): T {
    this.name = StringHolder(name)
    return this
}

@Deprecated("Please consider to replace with the actual property setter")
fun <T : Nameable> T.withName(name: StringHolder?): T {
    this.name = name
    return this
}